typescript/core/src/McapWriter.ts

```diff
@@ -120,7 +129,60 @@ export class McapWriter {
     this.#compressChunk = compressChunk;
   }
 
+  /**
+   * Initializes a new McapWriter in append mode.
+   */
+  static async InitializeInAppendMode({
+    writable,
+    readable,
+  }: {
+    writable: IAppendWritable;
+    readable: IReadable;
+  }): Promise<McapWriter> {
+    const reader = await McapIndexedReader.Initialize({ readable });
+
+    await writable.seek(reader.dataEndOffset!);
```

Comment:
Generally, it would be nice to avoid `!` and instead throw meaningful error messages if required fields are missing. Under what circumstances would this be undefined?